src/main/java/io/codelex/oop/Cars/CarService.java

```diff
@@ -0,0 +1,108 @@
+package io.codelex.oop.Cars;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CarService {
+
+    private List<Car> carList = new ArrayList<>();
+
+    public List<Car> getCarList() {
+        return carList;
+    }
+
+    public void addCar(Car car) {
+        carList.add(car);
+    }
+
+    public void removeFromCarList(Car car) {
+        carList.remove(car);
+    }
+
+    public List<Car> getCarByEngine(EngineType engineType) {
+        List<Car> carsWithSpecificEngineType = new ArrayList<>();
+        for (Car c : carList) {
+            if (c.getEngineType() == engineType) {
+                carsWithSpecificEngineType.add(c);
+            }
+        }
+        return carsWithSpecificEngineType;
+    }
+
+    public List<Car> getYearBefore(int year) {
+        List<Car> carsBeforeSpecificYear = new ArrayList<>();
+        for (Car c : carList) {
+            if (c.getYearOfManufacture() < year) {
+                carsBeforeSpecificYear.add(c);
+            }
+        }
+        return carsBeforeSpecificYear;
+    }
+
+    public void getMostExpensiveCar() {
+        Car car = Collections.max(carList, Comparator.comparingDouble(Car::getPrice));
+        System.out.println(car);
+    }
+
+    public void getCheapestCar() {
+        Car car = Collections.min(carList, Comparator.comparingDouble(Car::getPrice));
+        System.out.println(car);
+    }
+
+    public void getNumberOfManufacturers() {
+        for (Car c : carList) {
+            if (c.getManufacturerList().size() >= 3) {
+                System.out.println(c);
+            }
+        }
+    }
+
+    public List<Car> sortByNameAscending() {
```

Comment:
nevajag taisīt jaunu variabli
`toList()` jau to izveido, just retunr that

Reply:
Salabots!